addons/mail/static/src/models/composer/composer.js

```diff
@@ -406,12 +406,20 @@ function factory(dependencies) {
                     ]],
                 });
             }
-            this.update({
-                [this.activeSuggestedRecordName]: [[
-                    'link',
-                    this[this.extraSuggestedRecordsListName][this[this.extraSuggestedRecordsListName].length - 1]
-                ]],
-            });
+            if (this[this.extraSuggestedRecordsListName][this[this.extraSuggestedRecordsListName]]) {
```

Comment:
This condition makes no sense: you are using the array as a key for itself (I don't blame you, code is indeed unreadable), but anyway the condition you want is already above in the method. There's just missing a `return` at the end of the above `if`.